src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -115,10 +115,15 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging {
       }
     }
 
-    def abortActiveWorkflows(workflows: Seq[Workflow]) = {
-      Future.traverse(workflows.collect { case wf if wf.workflowId.isDefined => wf })(wf =>
-        Future.successful(wf.workflowId).zip(executionServiceCluster.abort(WorkflowExecution(wf), getUserInfo))
-      )
+    def abortActiveWorkflows(submissionId: UUID, workflows: Seq[Workflow]) = {
+      datasource.inTransaction { dataAccess =>
+        val extIds = workflows.collect { case wf if wf.workflowId.isDefined => wf.workflowId.get }
+        // look up WorkflowRecs for these workflows
+        val wrquery = dataAccess.workflowQuery.findWorkflowByExternalIdsAndSubmissionId( extIds, submissionId)
```

Comment:
rather than a potentially large in clause you can query directly for only the workflows you want to abort, i.e.
where submission_id = submissionId and external_id is not null and status in (running statuses)

I know that last condition was not in the original code but it probably should be, no need to call abort on workflows already done.